### DISCUSSION TITLE
increase line size

### DISCUSSION THREAD

**Discussion by Masber** [sourcelair/xterm.js#4488] [Q&A] [answered]
Hi,

I am pretty new to xterm.js, I have a demo and realised xterm wraps the text after 80 chars, how can I remove this limit? I would like xterm to use all the width available in the window for the terminal.

![image](https://user-images.githubusercontent.com/722064/233811452-06dcd9d4-3edd-42d8-88d3-493f5a3bc997.png)

thank you very much

**Comment by jerch** [comment]
The terminal instance has a `resize` method to change the grid size of the terminal. There is also the [fit addon](https://github.com/xtermjs/xterm.js/tree/master/addons/xterm-addon-fit) to align the terminal size with a certain div-container size.
Last but not least you need to propagate the changed grid size back to the pty and apply there as well (eg. with `node-pty` call `resize` there, too).

**Reply by Masber** [reply]
ok, sorry I'm pretty new to this, I am not sure why the resize with the node-pty, my backend is in rust hence I assume I need to find the equivalent of node-pty... is there any documentation you would recommend to understand how this communications/protocols works? so I can read and understand what I need from the server perspective and then try to find the right library?

**Reply by jerch** [reply, marked as answer]
Well good documentation about TTY/PTY is hard to find. Maybe one of these work for you: https://dev.to/napicella/linux-terminals-tty-pty-and-shell-192e or https://www.linusakesson.net/programming/tty/

`node-pty` is just a nodejs binding to the PTY interface of the OS, I am pretty sure rust has something like this as well.
